common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -632,6 +637,24 @@ def get_problem_html(self, encapsulate=True):
             'weight': self.weight,
         }
 
+        # Handle demand hints
+        demand_hints = self.lcp.tree.xpath("//problem/demandhint/hint")
+        show_hint_button = len(demand_hints) > 0
+        demand_hint = None  # if set to something, will show up in UI
+        if hint_index is None:
+            hint_index = 0  # bootstrap case, the client will start at 0
+        elif len(demand_hints) > 0:
```

Comment:
This is the only section of code that's untested, that makes me wonder if it should be. [coverage report](https://build.testeng.edx.org/job/edx-platform-all-tests-pr-flow/183/Diff_Coverage_Report/) for reference.